tmt/steps/provision/__init__.py

```diff
@@ -59,6 +60,254 @@ class CheckRsyncOutcome(enum.Enum):
     INSTALLED = 'installed'
 
 
+class GuestPackageManager(enum.Enum):
+    DNF = 'dnf'
+    YUM = 'yum'
+    RPM_OSTREE = 'rpm-ostree'
+
+
+T = TypeVar('T')
+_FactQuery = Callable[['GuestFacts', 'Guest'], Optional[T]]
+_SafeFactQuery = Callable[['GuestFacts', 'Guest'], T]
+
+
+def raise_on_none(family: str) -> Callable[[_FactQuery[T]], _SafeFactQuery[T]]:
+    def outer(fn: _FactQuery[T]) -> _SafeFactQuery[T]:
+        @functools.wraps(fn)
+        def inner(self: 'GuestFacts', guest: 'Guest') -> T:
+            fact = fn(self, guest)
+
+            if fact is not None:
+                return fact
+
+            raise tmt.utils.GeneralError(
+                f'Failed to identify {family} of guest "{guest}".')
```

Comment:
```suggestion
                f"Failed to identify {family} of guest '{guest}'.")
```

Consistent user [messaging](https://tmt.readthedocs.io/en/latest/contribute.html#introduction).

Reply:
It's not used, removed in https://github.com/teemtee/tmt/commit/07eca5234a494aa2f97185002790949807afbb84.

Reply:
Hopefully answered, I need to resolve this in order to be able to merge.